tdms/tests/system/data/input_generation/generate_test_input.py

```diff
@@ -0,0 +1,236 @@
+    run_bscan(input_arguments_in_the_correct_order).
+    """
+
+    # The directory of the test whose input data is being generated
+    test_directory: str
+    # The path to the input file that defines the variables iteratefdtd_matrix.m takes in
+    input_filename: Union[str, Path]
+
+    def __init__(self, test_directory, input_filename) -> None:
+        """Initialise by unpacking values, and assigning defaults if necessary."""
+        self.test_directory = test_directory
+        self.input_filename = input_filename
+        return
+
+    def run_bscan(self, engine: MatlabEngine) -> None:
+        """Runs the run_bscan function in the MatlabEngine provided.
+
+        The bscan/ and matlab/ directories are assumed to already be in the includepath of the engine instance, so that the run_bscan and supporting matlab files can be called.
+        """
+        # function [] = run_bscan(test_directory, input_filename)
+        # Cast to str() to guard against Path instances slipping through
+        engine.run_bscan(str(self.test_directory), str(self.input_filename), nargout=0)
+        return
+
+
+class MATLABEngineWrapper:
+    """When we regenerate input data, we always need to add the bscan/ and matlab/ directories to the MATLAB instance's search path. correspondingly, we always want to kill the MATLAB instance after running the run_bscan function and generating the data.
+
+    This class serves as a wrapper for that purpose. It stores instance(s) of the BScanArguments class, which it will run in sequence between the aforementioned addpath() setup and then engine shutdown. The .run() method performs exactly this.
+    """
+
+    # List containing the subsequent matlab commands to be executed by the interpreter
+    bscan_calls: list[BScanArguments]
+
+    # The MATLAB session that will run
+    engine: MatlabEngine
+
+    def __init__(self, bscans: Union[BScanArguments, list[BScanArguments]]) -> None:
+        """Initialise with the provided (list of) BScanArguments."""
+        self.bscan_calls = []
+        # add the additional commands here
+        if isinstance(bscans, BScanArguments):
+            # handed BScanArgument, construct command from this
+            self.bscan_calls = [bscans]
+        elif isinstance(bscans, list):
+            # validate this is a list of BScanArguments and not some other datatype
+            non_bscan_items = sum([isinstance(b, BScanArguments) for b in bscans])
+            if non_bscan_items > 0:
+                raise RuntimeError(
+                    f"Error: not all inputs are BScan calls ({non_bscan_items}/{len(bscans)})"
+                )
+            else:
+                self.bscan_calls = bscans
+        else:
+            # I don't know what we've been handed, but it's not what I was expecting
+            raise RuntimeError(
+                f"Error: expected BScanArguments or list[BScanArguments] but got {type(bscans)}"
+            )
```

Comment:
Past me agrees with you - this will eventually turn [into this](https://github.com/UCL/TDMS/blob/a0f1d5958cc67f9b7bb109148bcce5e6116d9368/tdms/tests/system/data/input_generation/generation_data.py#L35), which simply uses the config file to deduce how many times `bscan` needs to run.